### PR TITLE
Implement yaml representation for new entry resource

### PR DESCRIPTION
**Pull request by philandstuff**
As part of this change I've updated the DBSupport infrastructure to
allow setting explicit entry timestamps in test data.

### CODE REVIEW

File: src/test/resources/fixtures/single.yaml [hunk 1]
```diff
@@ -1,8 +1,4 @@
 ---
-serial-number: 1
-hash: "someHash1"
-entry:
-  fields:
-  - "field1"
-  register: "value1"
-  text: "The Entry 1"
+entry-number: "1"
```
Comment (philandstuff) [inline src/test/resources/fixtures/single.yaml:2]: Hold on a minute, I haven't worked out why this turned into a string.

Comment (om-sharma) [inline src/test/resources/fixtures/single.yaml:2]: Specification https://openregister.github.io/specification/#entry-resource specifies all field values as string.

Comment (philandstuff) [inline src/test/resources/fixtures/single.yaml:2]: ah okay, and https://openregister.github.io/specification/#json-representation says

> All field values MUST be encoded as JSON strings.

for the JSON representation. We should probably decide if we want to do the same for YAML or not.

Comment (philandstuff) [inline src/test/resources/fixtures/single.yaml:2]: I raised openregister/specification#51 for this issue. Let's just go with strings for the moment, we can change it later if necessary.

File: src/test/java/uk/gov/register/presentation/functional/testSupport/TestDAO.java [hunk 2]
```diff
@@ -15,6 +18,7 @@
     private TestDAO(String databaseName, String user) {
         String postgresConnectionString = String.format("jdbc:postgresql://localhost:5432/%s?user=%s", databaseName, user);
         DBI dbi = new DBI(postgresConnectionString);
+        dbi.registerArgumentFactory(new InstantArgumentFactory(Optional.empty()));
```
Comment (om-sharma) [inline src/test/java/uk/gov/register/presentation/functional/testSupport/TestDAO.java:21]: There is already a default constructor of InstantArgumentFactory.

**Pull request merged**